src/prometheus_alert_rules/prometheus_configuration_reload_failure.rule

```diff
@@ -0,0 +1,12 @@
+# Based on https://awesome-prometheus-alerts.grep.to/rules.html#prometheus-self-monitoring-1
```

Comment:
To prevent confusion, should the folder be named "self_monitoring_alert_rules"?

Reply:
Confusion for whom? Leaving it as the default path says "this charm provides alert rules following the same pattern as all other charms which do".

Since there's no magic here (it's just `MetricsEndpointProvider` eating its own tail), using the default seems logical to me here.